src/index.js

```diff
@@ -231,35 +305,37 @@ function download_video(req, res) {
  * @param  {Function} callback The function to execute about load completion
  */
 function wait_for_video(id, callback) {
-  setTimeout(function() {
-    request(heroku + "/alexa/v3/cache/" + id, function(err, res, body) {
-      if (!err) {
-        var body_json = JSON.parse(body);
-        if (body_json.downloaded) {
-          callback();
-        }
-        else {
-          wait_for_video(id, callback);
-        }
+  request(heroku + "/alexa/v3/cache/" + id, function(err, res, body) {
```

Comment:
I've changed it from "wait and check" to "check and wait" to save the few seconds in case the cache should be already ready (e.g. from previous searches).